### PR TITLE
Fixed adding/removing multiple types of metadata

### PR DESCRIPTION
**Pull request by sams96**
So my commit febe5dda9fc03b277d44dc5abe5de529456fe860 broke both adding and removing multiple metadata fields at the same time, this fixes that while still sanitising the inputs. Thanks to @phweyland for pointing out my mistakes.

### CODE REVIEW

File: src/common/metadata.c [hunk 1]
```diff
@@ -115,11 +116,18 @@ static void _bulk_add_metadata(GList *metadata_list)
...
     DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), query, -1, &stmt, NULL);
-    sqlite3_bind_text(stmt, 1, g_list_nth_data(metadata_list, 0), -1, 0);
-    sqlite3_bind_text(stmt, 2, g_list_nth_data(metadata_list, 1), -1, 0);
-    sqlite3_bind_text(stmt, 3, g_list_nth_data(metadata_list, 2), -1, 0);
+    for(int i = 0; i < g_list_length(metadata_list); i++)
+      sqlite3_bind_text(stmt, i + 1, g_list_nth_data(metadata_list, i), -1, 0);
+
     sqlite3_step(stmt);
     sqlite3_finalize(stmt);
   }
```
Comment (TurboGit) [inline src/common/metadata.c]: You need a g_free(query) here.

Comment (sams96) [reply src/common/metadata.c]: 🤦 Maybe I'll remember next time

**Comment by sams96** [comment]
Maybe hold off on merging for a minute to see if #4142 has something to do with this, and if I can fix it.

**Comment by TurboGit** [comment]
@sams96 : sure, ping me when you feel this ready. I'll mark as wip.

**Comment by sams96** [comment]
@TurboGit I think it's good to go now

**Comment by TurboGit** [comment]
Ok, thanks for fixing this regression.

**Pull request merged**